scripts/grt/export.py

```diff
@@ -280,41 +280,52 @@ def annotate(label, human_label=None):
         logging.debug("Processing %s:%s", offset_start, min(end_job_id, offset_start + args.batch_size))
-        for job in sa_session.query(model.Job) \
-                .filter(model.Job.id > offset_start) \
-                .filter(model.Job.id <= min(end_job_id, offset_start + args.batch_size)) \
-                .all():
-
-            # If the tool is blacklisted, exclude everywhere
-            if job.tool_id in blacklisted_tools:
-                continue
-
-            input_datasets=(job.input_datasets)
-            for input_dataset in input_datasets:
-                hda = input_dataset.dataset
-                dataset = hda.dataset
-                datasets=({'dataset_id': str(hda.dataset_id),
-                           'extension': hda.extension,
-                           'file_size': int(dataset.file_size),
-                           'name': hda.name,
-                           'param_name': input_dataset.name})
-
-                handle_datasets.write(str(job.id))
-                handle_datasets.write('\t')
-                handle_datasets.write(datasets['dataset_id'])
-                handle_datasets.write('\t')
-                handle_datasets.write(datasets['extension'])
-                handle_datasets.write('\t')
-                handle_datasets.write(str(datasets['file_size']))
-                handle_datasets.write('\t')
-                handle_datasets.write(datasets['name'])
-                handle_datasets.write('\t')
-                handle_datasets.write(datasets['param_name'])
-                handle_datasets.write('\n')
 
+        # three queries: JobToInputDatasetAssociation, HistoryDatasetAssociation, Dataset
+
+        job_to_hda_ids=sa_session.query(model.JobToInputDatasetAssociation.job_id, model.JobToInputDatasetAssociation.dataset_id,
+                model.JobToInputDatasetAssociation.name) \
+                .filter(model.JobToInputDatasetAssociation.job_id > offset_start) \
+                .filter(model.JobToInputDatasetAssociation.job_id <= min(end_job_id, offset_start + args.batch_size)) \
+                .all()
+
+        # put all of the hda_ids into a list
+        hda_ids=[i[1] for i in job_to_hda_ids]
+
+        hdas=sa_session.query(model.HistoryDatasetAssociation.id, model.HistoryDatasetAssociation.dataset_id, 
+                model.HistoryDatasetAssociation.extension) \
+                .filter(model.HistoryDatasetAssociation.id.in_(hda_ids)) \
+                .all()
+
+        # put all the dataset ids into a list
+        dataset_ids=[i[1] for i in hdas]
+
+        # get the sizes of the datasets
+        datasets=sa_session.query(model.Dataset.id, model.Dataset.total_size) \
+                .filter(model.HistoryDatasetAssociation.id.in_(dataset_ids)) \
+                .all()
+
+        # datasets to dictionay for easy search
+        hdas={i[0]:i[1:] for i in hdas}
+        datasets={i[0]:i[1:] for i in datasets}
+
+        for job in job_to_hda_ids:
```

Comment:
this should respect the tool blacklist

Reply:
ok

Reply:
arguably less of an issue? The tool blacklist was more to say "don't send parameters for this tool because might contain secrets".

Without knowing what tool it was, knowing the input files + their sizes is useless.

Reply:
there used to be a dataset name in the previous version, but this one is safe I think `'job_id', 'dataset_id', 'extension', 'file_size', 'param_name', 'type'`